sdk/azidentity/doc.go

```diff
@@ -0,0 +1,89 @@
+// +build go1.13
+
+// Copyright 2017 Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by an MIT
+// license that can be found in the LICENSE file.
+
+/*
+Package azidentity implements a set of credential types for use with
+Azure SDK clients that support Azure Active Directory (AAD) token authentication.
+
+The following credential types are included in this module:
+	- AuthorizationCodeCredential
+	- AzureCLICredential
+	- ChainedTokenCredential
+	- ClientCertificateCredential
+	- ClientSecretCredential
+	- DefaultAzureCredential
+	- DeviceCodeCredential
+	- EnvironmentCredential
+	- InteractiveBrowserCredential
+	- ManagedIdentityCredential
+	- UsernamePasswordCredential
+
+By default, the recommendation is that users call NewDefaultAzureCredential() which will
+provide a default ChainedTokenCredential configuration conformed of:
+	- EnvironmentCredential
+	- ManagedIdentityCredential
+	- AzureCLICredential
+Configuration options can be used to exclude any of the previous credentials from the
+DefaultAzureCredential implementation.
+
+Example call to NewDefaultAzureCredential():
+	cred, err := NewDefaultAzureCredential(nil) // pass in nil to get default behavior for this credential
+	if err != nil {
+		// process error
+	}
+	client := armresources.NewResourcesClient(armcore.NewDefaultConnection(cred, nil), "<subscription ID>")
+
+Example call to NewDefaultAzureCredential() with options set:
+	// these options will make sure the AzureCLICredential will not be added to the credential chain
+	opts := DefaultAzureCredentialOptions{ExcludeAzureCLICredential: true}
```

Comment:
Nit: typically this would be expressed as a struct literal at the call site.

Reply:
Done